### ISSUE TITLE
Problems with the [HideInEditor] attribute

### ISSUE THREAD

**Issue by PhyresiCompany** [FlaxEngine/FlaxEngine#780]
**Issue description:**
Using the **[HideInEditor]** attribute when entering the Start Game the variable has its value reset

**Steps to reproduce:**

https://user-images.githubusercontent.com/101017436/196801980-d6d8c849-ef8f-41a7-bbd3-7a9967de3da4.mp4

**Flax version:**
1.4

**Comment by PhyresiCompany** [comment]
> 

Yes, I used a variable **[ShowInEditor] private Actor** and when I gave Start Game by the editor the variable reset its value

**Comment by stefnotch** [comment]
Working as intended then.
https://docs.flaxengine.com/manual/scripting/serialization/index.html#serialization-rules
https://github.com/FlaxEngine/FlaxEngine/blob/f6857f66ba5ef2deb6bdeeccfc683a89896359fa/Source/Engine/Scripting/Attributes/Editor/ShowInEditorAttribute.cs#L11

But, this is something that trips up a lot of people. Any ideas on how to make this more obvious, or easier to figure out?

**Comment by PrecisionRender** [comment]
Unity automatically serializes fields when the user wants them shown in the editor. Perhaps renaming "SerializeAttribute" to "SerializeField" will make it more obvious to Unity users what it does, and when they don't see it in the editor, they should be able to easilyfind the "ShowInEditor" attribute.

**Comment by PhyresiCompany** [comment]
> Working as intended then. https://docs.flaxengine.com/manual/scripting/serialization/index.html#serialization-rules
> 
> https://github.com/FlaxEngine/FlaxEngine/blob/f6857f66ba5ef2deb6bdeeccfc683a89896359fa/Source/Engine/Scripting/Attributes/Editor/ShowInEditorAttribute.cs#L11
> 
> But, this is something that trips up a lot of people. Any ideas on how to make this more obvious, or easier to figure out?

Download my project and test if the same error happens to you [https://drive.google.com/file/d/12q9oBQVy5bQdJh8xf7-qcYsGP1EyyrhW/view?usp=sharing](url)

**Comment by PrecisionRender** [comment]
The message you got in the Debug window is because the variable value was not serialized and the code that references the Actor has no Actor to reference. Use this attribute: `[Serialize, ShowInEditor]`

**Comment by PhyresiCompany** [comment]
> The message you got in the Debug window is because the variable value was not serialized and the code that references an Actor using that variable has no Actor to reference, due to the variable being null. Use this attribute: `[Serialize, ShowInEditor]`

Thanks
